unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarLODController/AvatarLODController.cs

```diff
@@ -49,17 +83,57 @@ private void ResetMeshUVs()
             uvs[1].Set(1, 0);
             uvs[2].Set(0, 1);
             uvs[3].Set(1, 1);
-            mesh.uv = uvs;
+            impostorMesh.uv = uvs;
+        }
+
+        private void UpdateAvatarMeshes(int layer, bool enabledState)
+        {
+            int count = avatarRenderers.Count;
+            for (var i = 0; i < count; i++)
+            {
+                avatarRenderers[i].gameObject.layer = layer;
+                avatarRenderers[i].enabled = enabledState;
+            }
         }
 
         public void ToggleLOD(bool enabled)
         {
-            if (meshRenderer.gameObject.activeSelf == enabled)
+            if (impostorMeshRenderer.gameObject.activeSelf == enabled)
                 return;
 
-            meshRenderer.gameObject.SetActive(enabled);
+            impostorMeshRenderer.gameObject.SetActive(enabled);
+
+            UpdateAvatarMeshes(PhysicsLayers.defaultLayer, !enabled);
+        }
+
+        public void UpdateImpostorSnapshot()
+        {
+            UpdateAvatarMeshes(PhysicsLayers.characterPreviewLayer, true);
+
+            SetImpostorTexture(TakeSnapshot());
+
+            UpdateAvatarMeshes(PhysicsLayers.characterPreviewLayer, false);
+
+            lastSnapshotsUpdateTime = Time.timeSinceLevelLoad;
+        }
+
+        private Texture2D TakeSnapshot()
+        {
+            // Position snapshot camera next to the target avatar
+            snapshotCamera.gameObject.SetActive(true);
+            snapshotCamera.transform.SetParent(Camera.main.transform);
+            snapshotCamera.transform.localPosition = Vector3.zero;
+            snapshotCamera.transform.forward = (impostorMeshRenderer.transform.position - snapshotCamera.transform.position).normalized;
+            snapshotCamera.transform.position = impostorMeshRenderer.transform.position + -snapshotCamera.transform.forward * 2f;
+
+            snapshotCamera.Render();
+            RenderTexture.active = snapshotRenderTexture;
+            snapshotTex.ReadPixels(new Rect(0, 0, snapshotRenderTexture.width, snapshotRenderTexture.height), 0, 0);
+            snapshotTex.Apply();
```

Comment:
Oh, of course, this is what is killing the performance. We should find a way to directly render to the RT, and assign the RT to the LOD billboards materials directly. This optimization only has chances to work if we never create `Texture2D` objects dynamically, and of don't use the crazy expensive `ReadPixels` + `Apply` combo.

Reply:
Right, I didn't know we could use the `RenderTexture` directly in the material and avoid the `Texture2D` update with `ReadPixels()` + `Apply()`.
I just updated the code to use 1 `RenderTexture` per avatar and no `Texture2D`, instead of 1 `Texture2D` per avatar and a shared `RenderTexture`.

Next I'll add the `Animation` component toggle, the shared `RenderTexture` atlas (without actual atlassing logic, just sharing the same renderTex) and after that we can join all our optimizations into an experimental branch to measure performance.